MantidQt/RefDetectorViewer/src/RefMatrixWSImageView.cpp

```diff
@@ -48,7 +48,7 @@ RefMatrixWSImageView::RefMatrixWSImageView(QString wpsName, int peakMin,
 
   std::vector<float> data(static_cast<size_t>(totalYMax) * sz);
 
-  for (size_t px = 0; px < totalYMax; px++) {
+  for (size_t px = 0; px < static_cast<size_t>(totalYMax); px++) {
```

Comment:
I think you are better off making `totalYMax` type `size_t` and then casting it when constructing `source`. 

Two other tiny things:
1. change `px++` to `++px`
2. use `boost::make_shared` to construct `source`.